polynote-kernel/src/main/scala/polynote/kernel/interpreter/scal/ScalaCompleter.scala

```diff
@@ -18,87 +21,46 @@ class ScalaCompleter[Compiler <: ScalaCompiler](
+        val isErrorOrEmpty = name.decoded == "<error>" || name.isEmpty
+        effectBlocking {
 
-        val enrichSelected = if (ownerTpe != null) {
-          effect(new analyzer.ImplicitSearch(qual, definitions.functionType(List(ownerTpe), definitions.AnyTpe), isView = true, context0 = context).allImplicits).map {
-            allImplicits =>
-              allImplicits.foreach {
-                result =>
-                  val members = result.tree.tpe.finalResultType.members
-                  members.foreach(scope.enterIfNew)
-              }
-          }
-        } else ZIO.unit
+          val context = locateContext(position).getOrElse(NoContext)
+          val fromCompiler = typeCompletions(context, sel)
 
-        enrichSelected.map {
-          _ => scopeToCompletions(scope, name, ownerTpe)
+          fromCompiler.results.filter {
+            result => result.accessible && (isErrorOrEmpty || result.sym.name.startsWith(name))
+          }
+        }.map {
+          results =>
+            results.map(memberToCompletion).distinct
         }
 
-      case _ => ZIO.succeed(Nil)
+      case _ =>
+        ZIO.succeed(Nil)
     }
 
-    def completeIdent(tree: Ident) = tree match {
-      case Ident(name) =>
-        val nameString = name.toString
-        val context = locateContext(tree.pos).getOrElse(NoContext)
-
-        val imports = context.imports.flatMap {
-          importInfo => importInfo.allImportedSymbols.filter(sym => !sym.isImplicit && sym.isPublic && sym.name.startsWith(name) && !sym.name.startsWith("deprecated"))
-        }
-
-        index.findMatches(nameString).flatMap {
-          fromClasspath =>
-            ZIO {
-              val cellScope = context.owner.info.decls.filter(isVisibleSymbol).filter(_.name.startsWith(name)).toList.map(_.accessedOrSelf).distinct
-
-              // somehow we're not getting the imported stuff in the context
-              val prevCellScopes = if (context.owner.isClass) {
-                context.owner.asClass.primaryConstructor.paramLists.headOption.toList.flatten.collect {
-                  case sym if sym.name.startsWith("_input") => sym.info.nonPrivateDecls.collect {
-                    case decl if (decl.isClass || decl.isMethod || decl.isType) && decl.name.startsWith(name) => decl
-                  }
-                }.flatten
-              } else Nil
-
-              val indexCompletions = fromClasspath.filterKeys(name => !imports.exists(_.nameString == name)).toList.flatMap {
-                case (simpleName, qualifiedNames) => qualifiedNames.map {
-                  case (priority, qualifiedName) => priority -> Completion(simpleName, Nil, Nil, qualifiedName, CompletionType.ClassType, Some(qualifiedName))
-                }
-              }.sortBy(_._1).take(10).map(_._2)
-              (cellScope ++ prevCellScopes ++ imports).map(symToCompletion(_, NoType)) ++ indexCompletions
-            }
-        }
-    }
+    def completeIdent(tree: Ident) =
```

Comment:
This was simplified quite a bit, because we can now delegate it to the presentation compiler with quite good results